.github/workflows/claude-code-review.yml

```diff
@@ -0,0 +1,138 @@
+name: Claude Code Review
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
+
+permissions:
+  contents: read
+  pull-requests: write
+  issues: read
+  id-token: write
+  actions: read
+
+jobs:
+  claude-review:
+    if: github.event.pull_request.user.login == 'squadgazzz'
+
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Run Claude Code Review
+        uses: anthropics/claude-code-action@v1
+        env:
+          GH_TOKEN: ${{ github.token }}
+        with:
+          claude_code_oauth_token: ${{ secrets.CLAUDE_CODE_OAUTH_TOKEN }}
+          prompt: |
+            REPOSITORY: ${{ github.repository }}
+            PR NUMBER: ${{ github.event.pull_request.number }}
+            PR AUTHOR: ${{ github.event.pull_request.user.login }}
+            PR TITLE: ${{ github.event.pull_request.title }}
+
+            You are a Rust code reviewer for the CoW Protocol services codebase.
+
+            TASK:
+            1. Generate a PR description if the PR body is empty
+            2. Review the PR diff and leave inline comments on problematic code
+
+            PART 1 - PR DESCRIPTION (only if PR body is empty):
+            Use `gh pr view ${{ github.event.pull_request.number }} --json body` to check if PR has a description.
+            If body is empty, generate and update PR description using:
+            ```
+            gh pr edit ${{ github.event.pull_request.number }} --body-file /dev/stdin <<'EOF'
+            ## Description
+            [High-level description of changes, background, and motivation]
+
+            ## Changes
+            - [List of specific changes made]
+            - [Each bullet should be a concrete change]
+
+            ## How to test
+            [How these changes should be tested. If no new tests were added, write "Existing tests"]
+
+            ## Related issues
+            <!-- Link related issues here, e.g., Fixes #123 -->
+            EOF
+            ```
+
+            PART 2 - CODE REVIEW:
+
+            REVIEW FOCUS (issues only, no positive feedback):
+            1. **Bugs & Logic Errors**: Race conditions, off-by-one errors, incorrect error handling, panic risks
```

Comment:
Add possible deadlocks (such as holding a regular mutex over an await), unexpected drops (like the request id issue)

Reply:
Yeah, I think deadlocks as a category would be helpful.